### ISSUE TITLE
no-string-literal to allow option for private member access

### ISSUE THREAD

**Issue by jgornick** [palantir/tslint#3674]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.6.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class Foo {
  public bar: boolean = false;
  private baz: boolean = false;
}

const foo = new Foo();
foo.bar = true; // access public
foo['baz'] = true; // access private
```

with `tslint.json` configuration:

```json
{
  "extends": [
    /* @see https://github.com/palantir/tslint/blob/master/src/configs/recommended.ts */
    "tslint:recommended"
  ],
  "rules": {
    // TSLint Overrides/Custom
    "no-string-literal": true
  }
}

```

#### Actual behavior

```
ERROR: (no-string-literal) foo.ts[9, 5]: object access via string literals is disallowed
```

#### Expected behavior

I would like to have an option that allows object access via string literals if access is to a private member.

**Comment by michaeljota** [comment]
I understand the rationale of this rule, but sometimes we want to enforce the usage of dot notation instead of the bracket one. If `--noImplicitAny` is activated, we could turn off this rule safely, but still the enforcement of using dot notation is loss.

**Comment by jgornick** [comment]
Isn't it fair to say though that the lint rule could look at the access of the field referenced to determine if it's `public`, `protected`, or `private`?

Just to be clear too, this would be an option not changing the `no-string-literal` default implementation.

**Comment by jgornick** [comment]
@JoshuaKGoldberg Totally makes sense and I don't really have a good reason to go against established practices for using member access.

The only reason why this was brought up because we were trying to access `protected` and/or `private` members during tests.

Thanks!
